examples/master/README.md

```diff
@@ -0,0 +1,102 @@
+# Jenkins Master example
+
+This directory contains an example of Jenkins setup, that is configured to
+demonstrate the CI/pipeline workflow for the sample-app application using the
+Jenkins Master/Slave setup and automatization done on OpenShift v3.
+
+<p align="center">
+<img width="420" src="https://raw.githubusercontent.com/mfojtik/jenkins-ci/master/jenkins-flow.png"/>
+</p>
+
+## Installation
+
+To start, you have to manually enter following commands in OpenShift:
+
+
+```console
+# Create project and allow Jenkins to talk to OpenShift API
+$ oc new-project ci
+$ oc policy add-role-to-user edit system:serviceaccount:ci:default
+
+# Create the 'staging' project where we deploy the sample-app for testing
+$ oc new-project stage
+$ oc policy add-role-to-user edit system:serviceaccount:stage:default
+$ oc policy add-role-to-user edit system:serviceaccount:ci:default
+
+# Now create the templates in the 'ci' and 'stage' projects
+$ oc create -f openshift -n ci
+$ oc create -f https://raw.githubusercontent.com/openshift/jenkins/master/examples/master/jenkins-master-with-k8s-plugin.json
+$ oc create -f https://raw.githubusercontent.com/openshift/jenkins/master/examples/slave/s2i-slave-template.json
+```
+
+## Instantiating templates from OpenShift web console
+
+Navigate to the OpenShift UI and choose the `ci` project we created in previous
+step. Now click on *Add to Project* button and then click on *Show All
+Template*. You should see *jenkins-master* template and *s2i-to-jenkins-slave*.
+
+Note: To continue, you have to first create your Jenkins Slave image. Please see
+the [instructions](../slave/README.md) here.
+
+### Jenkins Master
+
+This template defines a
+[BuildConfig](https://docs.openshift.org/latest/dev_guide/builds.html#defining-a-buildconfig)
+that again, use the [Docker
+Strategy](https://docs.openshift.org/latest/dev_guide/builds.html#docker-strategy-options)
+to rebuild the official [OpenShift Jenkins Image](https://github.com/openshift/jenkins).
+This template also defines a [Deployment Configuration](https://docs.openshift.org/latest/dev_guide/deployments.html#creating-a-deployment-configuration) that will start just one instance
+of the Jenkins server.
+
+When you choose the `jenkins-master` template, you have to specify these parameters:
+
+* **JENKINS_SERVICE_NAME** - The name of the Jenkins service (default: *jenkins*)
+* **JENKINS_IMAGE** - The name of the original Jenkins image to use
+* **JENKINS_PASSWORD** - The Jenkins 'admin' user password
```

Comment:
I see, to parametrize the build. Gotcha and yes, that is very good idea. Do you also want to parametrize the 'git ref' ? It might help testing ;-)